components/inputs/input-date.js

```diff
@@ -438,13 +440,16 @@ class InputDate extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(LitEl
 	_handleKeydown(e) {
 		// open dropdown on down arrow or enter and focus on calendar focus date
 		if (e.keyCode === 40 || e.keyCode === 13) {
+			this._openedOnKeydown = true;
 			this.opened = true;
 			e.preventDefault();
 		}
 	}
 
-	_handleMouseup() {
+	_handleMouseup(e) {
 		this._inputTextFocusMouseup = true;
+		const icon = this.shadowRoot.querySelector('d2l-icon');
+		this._openCalendarOnly = e.srcElement.tagName === icon.tagName;
```

Comment:
Doesn't seem necessary to get the icon using its tag name and then compare the tag name...

```suggestion
		this._openCalendarOnly = e.srcElement.tagName === 'd2l-icon';
```